server/executor/runner.go

```diff
@@ -65,6 +68,10 @@ func (r persistentRunner) handleDBError(err error) {
 }
 
 func (r persistentRunner) Start(workers int) {
+	r.tp.OnPollComplete(func(result openapi.TestRunResult) {
```

Comment:
🚀  This means that on every trace poll end we are going to `try` to run the assertions right? in case the assertion contains the `Selector` entry

Reply:
I don't think of the runner as an orchestrator. The whole process is a chain of queues. I would move this dependency to the `tracePoller`, so when that queue determines the trace is `polled`, or ready to be asserted, it can just pass it to the assertion runner, much in the same way this queue sends the work to the next queue